luda-editor/server/server-core/src/apis/project/edit_user_acl.rs

```diff
@@ -33,21 +33,35 @@ pub async fn edit_user_acl(
     match permission {
         Some(permission) => crate::dynamo_db()
             .transact()
-            .update_item(UserInProjectAclDocumentUpdate {
-                pk_project_id: project.id,
+            .update_or_create_item(UserInProjectAclDocumentUpdateOrCreate {
+                pk_project_id: project_id,
                 sk_user_id: user_id,
-                update: move |mut document: UserInProjectAclDocument| async move {
+                update: move |mut document| async move {
                     document.permission = permission;
                     Ok(document)
                 },
+                create: move || async move {
+                    Ok(UserInProjectAclDocument {
+                        user_id,
+                        project_id,
+                        permission,
+                    })
+                },
             })
-            .update_item(ProjectAclUserInDocumentUpdate {
-                pk_user_id: session.user_id,
-                sk_project_id: project.id,
-                update: move |mut document: ProjectAclUserInDocument| async move {
+            .update_or_create_item(ProjectAclUserInDocumentUpdateOrCreate {
+                pk_user_id: user_id,
+                sk_project_id: project_id,
+                update: move |mut document| async move {
                     document.permission = permission;
                     Ok(document)
                 },
+                create: move || async move {
+                    Ok(ProjectAclUserInDocument {
```

Comment:
Other thing seems really good but would you remove Result type for create part? 
Update needs Result because it should check saved data after get item. 
But create doesn't.

Reply:
Hmm... wait. maybe future + Result for create would be make sence... right? how do you think?

Reply:
I don't know. I just used this for flexibility.

I thought `Create` could be async and maybe fail, not just the synchronous constructor.
But I didn't find this case yet.

So I think it's also okay to remove result type for `create`

Reply:
Let it be Future<Result>. I think it's right.